src/vehicle.h

```diff
@@ -734,24 +741,32 @@ class vehicle
         vehicle_part_range get_parts() const;
         /**
          * Yields a range of parts of this vehicle that each have the given feature
-         * and are not broken. Removed parts are also excluded. The enabled status
-         * of the part is ignored.
+         * and are available: not broken, removed, or part of a carried vehicle.
+         * The enabled status * of the part is ignored.
          */
         /**@{*/
-        vehicle_part_with_feature_range<std::string> get_parts( std::string feature ) const;
-        vehicle_part_with_feature_range<vpart_bitflags> get_parts( vpart_bitflags f ) const;
+        vehicle_part_with_feature_range<std::string> get_avail_parts( std::string feature ) const;
+        vehicle_part_with_feature_range<vpart_bitflags> get_avail_parts( vpart_bitflags f ) const;
         /**@}*/
         /**
-         * Like @ref get_parts, but also yields broken parts.
+         * Like @ref get_avail_parts, but also yields parts on carried vehicles.
          */
         /**@{*/
-        vehicle_part_with_feature_range<std::string> get_parts_including_broken(
+        vehicle_part_with_feature_range<std::string> get_parts_including_carried(
             std::string feature ) const;
-        vehicle_part_with_feature_range<vpart_bitflags> get_parts_including_broken(
+        vehicle_part_with_feature_range<vpart_bitflags> get_parts_including_carried(
             vpart_bitflags f ) const;
         /**@}*/
         /**
-         * Like @ref get_parts, but only yield enabled parts. They must still be unbroken.
+         * Like @ref get_avail_parts, but also yields broken parts and parts on carried vehicles.
+         */
+        /**@{*/
+        vehicle_part_with_feature_range<std::string> get_all_parts( std::string feature ) const;
+        vehicle_part_with_feature_range<vpart_bitflags> get_all_parts( vpart_bitflags f ) const;
```

Comment:
Out of curiosity what is the difference between `get_parts` and `get_all_parts`. Judging by the name, the former seems to yield less parts than the later (`get_parts` returns "not all part", if it did, why would it be named different from `get_all_parts`).

Reply:
Renamed to get_any_parts for "gets any part with a feature", get_all_parts for "gets all the parts on a vehicle" and get_parts_at for "get parts at pos."